### PR TITLE
Require qt < 5 until traitsui adds support for qt5

### PR DESCRIPTION
**Pull request by francisco-dlp**
Currently HyperSpy failts to start with the qt backend due to the current incompatibility between traitsui and Qt5, see hyperspy/hyperspy#1323.

### CODE REVIEW

**Comment by francisco-dlp** [comment]
@to266, any idea of why this is failing?

**Comment by francisco-dlp** [comment]
@to266, the notebook interaction issue was due to a ipykernel bug and I've fixed it by skipping the 2 buggy versions. The cythonize test is still failing.

**Comment by sem-geologist** [comment]
ow gawd the circleci is so uber ugly!

ok, I think I know there the problem lies (it is hard to notice due to:```placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder_placeholder....```

thing is the source should be built with --inplace flag before tests, same as in appveyor and travis

**Comment by sem-geologist** [comment]
exactly such line should land in the script before tests starts:
```python setup.py build_ext --inplace```

**Comment by francisco-dlp** [comment]
Great, thanks! Wouldn't it be better then to add a `setup.cfg` file with `inplace=1`?

**Comment by sem-geologist** [comment]
what is setup.cfg? is this circleci specific configuration?

**Comment by francisco-dlp** [comment]
No, it's a distutils feature. It sets the default to `--in-place` when building the extension. Is that what we want?

**Comment by sem-geologist** [comment]
--in-place is used by pip development installation, automatically, then somebody installs it normaly we do not want it to be built in place, normally it is built in separate built folder and then after copied to user or system python directory during installation. I think the problem is actually why tests can't use the hyperspy version which just got installed to the container, why it tries to run the tests on the source tree. due to that we need to build inplace.

**Comment by sem-geologist** [comment]
this is weird, run_test.py is not run on py35 tests (line 4149 in the log)

**Comment by sem-geologist** [comment]
looks py34 (line 1938 in the log) have the same problem, but how cython does not fail there??

**Comment by francisco-dlp** [comment]
It is actually not running for py34 either (L1938), the issue seems to be that the script is not executed in the hyperspy folder. In any case, I get the feeling that we're not fixing this in the right way: I've just checked the scikit learn recipe and they don't need to compile the cython extension inplace...

File: recipe/run_test.py [hunk 1]
```diff
@@ -6,6 +6,9 @@
     from nose import run_exit
     from traits.etsconfig.api import ETSConfig
     import os
+    from subprocess import call
+    call("python setup.py build_ext --inplace",
```
Comment (sem-geologist) [inline recipe/run_test.py]: maybe os.path.join(filepath,'..','setup.py') should be constructed...
maybe call works from the path of file?

Comment (francisco-dlp) [inline recipe/run_test.py]: I think that you're right. Still,  shouldn't we try to find the real issue with py35?

Comment (sem-geologist) [inline recipe/run_test.py]: I see what scikit learn does different than us:
they install thingy using `python setup.py develop` which compiles and uses everything in place, so they dont need to build things twice: for global user path and for tests

Comment (francisco-dlp) [inline recipe/run_test.py]: How did you come to this conclusion? And, would you suggest that we go that way?

Comment (sem-geologist) [inline recipe/run_test.py]: I came to this conclusion as I read source of scikit-learn now:
https://github.com/scikit-learn/scikit-learn/blob/master/build_tools/circle/build_doc.sh#125
https://github.com/scikit-learn/scikit-learn/blob/master/build_tools/travis/install.sh#L117

I wish I had learned about this earlier...

**Comment by francisco-dlp** [comment]
The funny thing about this is that it works in our own travis test, it only fails here... @sem-geologist, what would you suggest to do?

**Comment by sem-geologist** [comment]
I am wondering what those do in scikit-image:
https://github.com/conda-forge/scikit-learn-feedstock/blob/master/recipe/meta.yaml#L43-L47

**Comment by sem-geologist** [comment]
also there is not a single 'develop' word in scikit-learn-feedstock, so they manage this somehow differently

**Comment by sem-geologist** [comment]
there is only two items under build in meta.yaml,
however in scikit-image I see much more,
cython and numpy I think is really needed in our case (numpy for numpy headers I guess as it is natively reached in cython)
https://github.com/conda-forge/scikit-learn-feedstock/blob/master/recipe/meta.yaml#L22-L30

**Comment by francisco-dlp** [comment]
ok, I'll try adding numpy.

**Comment by sem-geologist** [comment]
and cython

**Comment by francisco-dlp** [comment]
cython is already there in this branch, that was my first (wrong) guess...

**Comment by sem-geologist** [comment]
what those two do?
https://github.com/conda-forge/hyperspy-feedstock/blob/master/recipe/meta.yaml#L13-L14

**Comment by francisco-dlp** [comment]
build is a tag to differentiate new builds of the same version. skip py2k is to skip building for py2 I guess.

By the way, cicleci keeps on failing :(

**Comment by francisco-dlp** [comment]
The good news is that the last two failures were due to the known random samfire failure issue, so there is still some hope...

**Comment by francisco-dlp** [comment]
Also, it is interesting to note that the same recipe was building the same v1.1.1 version before, so something must have changed somewhere else, maybe a conda bug? Or a new (buggy) cython version?

**Comment by francisco-dlp** [comment]
@sem-geologist, I have asked for help in the [conda-forge gitter chat](https://gitter.im/conda-forge/conda-forge.github.io?at=58762633dec171b811d50b1f) but I didn't get any help yet. I've checked other python packages containing Cython extensions and they all seem to be working fine, so I guess that there is something specific to our code that makes it fail...

**Comment by francisco-dlp** [comment]
Well, may it be the test itself what is faulty? It's checking if the path  `/cython/test_cython_integration.c` exists but, maybe the file is deleted after compilation in py35?

**Comment by francisco-dlp** [comment]
Indeed the actual Cython extensions seem to be running fine.

**Comment by isuruf** [comment]
It's not deleted after compilation, but it is not installed. Tests are run in a clean environment, so the if the build artifacts were not installed, they will not be found.
What's weird is that since Cython is not installed at run-time nor at test-time, that test should have been skipped.

**Comment by francisco-dlp** [comment]
Thanks @isuruf, that makes sense. Actually the test was probably put in place by @sem-geologist for when running in develop mode. It is indeed skipped in Py34 as it should, but not in Py35.

**Comment by sem-geologist** [comment]
why cython is not installed during run-time or test-time?

**Comment by francisco-dlp** [comment]
We forgot to update the recipe after you included the first cython extension in the project in v1.0... but everything seemed to work fine. I've just included in the PR Cython for building but, why should we need it for running?

**Comment by sem-geologist** [comment]
looking to the test, cython is present during test, else the test would be skipped, so cython library itself it is not the problem, the problem is that ... wait the file looks like wtf??? it have no line to launch the actual test, instead it  checks for c file (what is wrong...., especially in such case when source and installed library path is different... I think this is bug... I am sorry... but I think at least we found the problem.

**Comment by francisco-dlp** [comment]
No worries: we should have checked the test first but that's also the last thing that I did after I tried all sort of stupid things :(

**Comment by sem-geologist** [comment]
The test is pointless, so maybe lets remove it completely, the fast_bcf is enought verboise on its own to inform if something is not well in building cython extensions. the test was added before fast_bcf, so the whole newly added cythonization to setup.py could be tested.
What do you think, could we simple rid of it completely?

**Comment by francisco-dlp** [comment]
+10^10 ;)

**Comment by sem-geologist** [comment]
done, I did it on RELEASE_next_patch branch

**Comment by francisco-dlp** [comment]
I am closing this as the bcf issue cannot be fixed here and the qt5 was fixed in #4.

**Comment by sem-geologist** [comment]
bcf is also having issue?

**Comment by francisco-dlp** [comment]
Sorry, I meant the cyhonize test.

**Pull request closed without merging**